### PR TITLE
Get generator supports nulls for boxed types

### LINKED ISSUE #641: Support nulls for boxed types in get resolvers made by Annotation Processor
There is a class:
```
class Task {
    @StorIOSQLiteColumn(name = "c_start_time")
    protected Long startTime;
}
```
In the database c_start_time is nullable.

Now generates:
```
object.startTime = cursor.getLong(cursor.getColumnIndex("c_start_time"));
```

Right:
```
if (!cursor.isNull(cursor.getColumnIndex("c_start_time"))) 
    object.startTime = cursor.getLong(cursor.getColumnIndex("c_start_time"));
```

### PR DESCRIPTION
**Pull request by nikitin-da**
Closes #641

### CODE REVIEW

**Comment by simtse** [comment]
What is the status of the Pull request? Can it be merged?

File: storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/introspection/JavaType.java [hunk 1]
```diff
@@ -59,4 +59,18 @@ public static JavaType from(@NotNull TypeMirror typeMirror) {
             throw new IllegalArgumentException("Unsupported type: " + typeMirror);
         }
     }
+
+    public boolean isBoxedType() {
```
Comment (artem-zinnatullin) [inline storio-common-annotations-processor/src/main/java/com/pushtorefresh/storio/common/annotations/processor/introspection/JavaType.java]: nice

File: storio-content-resolver-annotations-processor/src/main/java/com/pushtorefresh/storio/contentresolver/annotations/processor/generate/GetResolverGenerator.java [hunk 2]
```diff
@@ -99,7 +99,16 @@ private MethodSpec createMapFromCursorMethodSpec(@NotNull StorIOContentResolverT
                 throw new ProcessingException(columnMeta.element, "Can not generate GetResolver for field");
             }
 
+            final boolean isBoxed = javaType.isBoxedType();
+            if (isBoxed) { // otherwise -> if primitive and value from cursor null -> fail early
+                builder.beginControlFlow("if(!cursor.isNull($L))", columnIndex);
```
Comment (artem-zinnatullin) [inline storio-content-resolver-annotations-processor/src/main/java/com/pushtorefresh/storio/contentresolver/annotations/processor/generate/GetResolverGenerator.java]: oh

File: storio-content-resolver-annotations/src/main/java/com/pushtorefresh/storio/contentresolver/annotations/StorIOContentResolverColumn.java [hunk 3]
```diff
@@ -23,4 +23,11 @@
      * @return true if column is key, false otherwise
      */
     boolean key() default false;
+
+    /**
+     * Optional: indicates that field should not be serialized when it hasn't value
```
Comment (artem-zinnatullin) [inline storio-content-resolver-annotations/src/main/java/com/pushtorefresh/storio/contentresolver/annotations/StorIOContentResolverColumn.java]: when its value is `null`?

File: storio-sqlite-annotations/src/main/java/com/pushtorefresh/storio/sqlite/annotations/StorIOSQLiteColumn.java [hunk 4]
```diff
@@ -26,4 +26,11 @@
      * @return true if column is key, false otherwise
      */
     boolean key() default false;
+
+    /**
+     * Optional: indicates that field should not be serialized when it hasn't value
```
Comment (artem-zinnatullin) [inline storio-sqlite-annotations/src/main/java/com/pushtorefresh/storio/sqlite/annotations/StorIOSQLiteColumn.java]: when its value is `null`

**Comment by artem-zinnatullin** [comment]
👍

**Comment by rsinukov** [comment]
👍

**Pull request merged**